packages/core/src/Page/index.js

```diff
@@ -567,7 +567,7 @@ class Page {
 
     const renderedHeader = this.pageConfig.variableProcessor.renderSiteVariables(this.pageConfig.sourcePath,
                                                                                  headerContent, pageSources);
-    return `${renderedHeader}\n${pageData}`;
```

Comment:
the newlines are important to ensure the different segments get their markdown rendered correctly

Reply:
Ah, I see! Have added the newline back to the code.